### PR TITLE
Add container instance id as a placeholder

### PR DESCRIPTION
**Pull request by JohnWu20**
**Description:** 

AWS ECS container insights instance level metrics use ContainerInstanceId for the log stream name, so this pr is to add container instance id as a placeholder.

Allow to use ContainerInstanceId as placeholder in log group/stream name.

Add unit test for ContainerInstanceId placeholder.

Update tests and readme

### CODE REVIEW

File: exporter/awsemfexporter/util.go [hunk 1]
```diff
@@ -26,9 +26,10 @@ import (
 )
 
 var patternKeyToAttributeMap = map[string]string{
-	"ClusterName": "aws.ecs.cluster.name",
-	"TaskId":      "aws.ecs.task.id",
-	"NodeName":    "k8s.node.name",
+	"ClusterName":         "aws.ecs.cluster.name",
+	"TaskId":              "aws.ecs.task.id",
+	"NodeName":            "k8s.node.name",
+	"ContainerInstanceId": "aws.ecs.container.instance.id",
```
Comment (hossain-rayhan) [inline exporter/awsemfexporter/util.go:32]: I think instance is not directly coupled to ECS. I don't think prefixing `instance.id` with `aws.ecs` is meaningful. Same goes for cluster though.

File: exporter/awsemfexporter/README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ The following exporter configuration parameters are supported.
...
+| `log_stream_name` | Customized log stream name which supports `{TaskId}`, `{ClusterName}`, `{NodeName}` and '{ContainerInstanceId}' placeholders. One valid example is `{TaskId}`. It will search for `TaskId` (or `aws.ecs.task.id`) resource attribute in the metrics data and replace with the actual task id. If none of them are found in the resource attribute map, `{TaskId}` will be replaced by `undefined`. Similar way,, for the `{ClusterName}`, it searches for `ClusterName` (or `aws.ecs.cluster.name`). For `{NodeName}`, it searches for `NodeName` (or `k8s.node.name`). For `{ContainerInstanceId}`, it searches for `ContainerInstanceId` (or `aws.ecs.container.instance.id`).                                             |"otel-stream"|
```
Comment (hossain-rayhan) [inline exporter/awsemfexporter/README.md:18]: How does `ContainerInstanceId` behave for ECS Fargate? Do we want to add a note that this feature only works for EC2 customers?

File: exporter/awsemfexporter/util.go (hunk 1 above)
Comment (JohnWu20) [reply exporter/awsemfexporter/util.go:32]: It makes sense, maybe it's better to use `aws.container.instance.id`. 
Also for the cluster, it seems that the name is defined by you, do you want me to change it?

Comment (hossain-rayhan) [reply exporter/awsemfexporter/util.go:32]: Oh my bad. If it's ECS Container Instance ID, then the name is fine. For EKS, I believe its NodeName.

**Review by hossain-rayhan — APPROVED** [review]
LGTM except the minor comment on README.

File: exporter/awsemfexporter/util.go (hunk 1 above)
Comment (JohnWu20) [reply exporter/awsemfexporter/util.go:32]: Got it, thanks

**Review by anuraaga — APPROVED** [review]

**Pull request merged**